### PR TITLE
feat(AWS Lambda): Improved function deployment messaging

### PR DESCRIPTION
**Pull request by rmidhun23**
The changes included in this PR aims at improving developer experience by adding additional messaging to AWS Lambda function deployments as discussed in #11614.

Closes: #11614

### CODE REVIEW

File: lib/plugins/aws/deploy-function.js [hunk 1]
```diff
@@ -392,8 +392,14 @@ class AwsDeployFunction {
 
     if (!Object.keys(_.omit(params, 'FunctionName')).length) {
       if (this.options['update-config']) log.notice();
+      const noticeMessage = [
+        'Configuration did not change.',
+        ' Configuration update skipped.',
```
Comment (medikoo) [inline lib/plugins/aws/deploy-function.js:397]: Maybe lest's reword it to:

```
Function configuration did not change, and the update was skipped.
If you made changes to the service configuration and expected them to be deployed, it most likely means that they can only be applied with a full service deployment.
```

**Review by medikoo — CHANGES_REQUESTED** [review]
@rmidhun23 Thank you! I've proposed some improvements to wording

File: lib/plugins/aws/deploy-function.js (hunk 1 above)
Comment (rmidhun23) [reply lib/plugins/aws/deploy-function.js:397]: Sure, will get back with the changes, thanks for the feedback.

Comment (rmidhun23) [reply lib/plugins/aws/deploy-function.js]: @medikoo I have incorporated the changes as requested.

**Review by medikoo — APPROVED** [review]
Thank you 👍
